src/test/storage/table_test.cpp

```diff
@@ -71,4 +72,11 @@ TEST_F(StorageTableTest, GetColumnIdByName) {
 
 TEST_F(StorageTableTest, GetChunkSize) { EXPECT_EQ(t.max_chunk_size(), 2u); }
 
+TEST_F(StorageTableTest, CompressChunkReplacesWithDictionarySegment) {
+  t.compress_chunk(ChunkID{0});
+  auto& chunk = t.get_chunk(ChunkID{0});
+  auto segment = chunk.get_segment(ColumnID{0});
+  EXPECT_NO_THROW(std::dynamic_pointer_cast<std::shared_ptr<DictionarySegment<int>>>(segment));
```

Comment:
std::dynamic_pointer_cast behaves differently than what I expected. It will return an empty shared pointer when the cast fails
-> do:
```
const auto value_segment = std::dynamic_pointer_cast<std::shared_ptr<DictionarySegment<int>>>(segment);
EXPECT_TRUE(value_segment, "compress_chunk did not replace ValueSegment with DictionarySegment");
```

Reply:
I think you should call it `dictionary_segment` or `segment` instead `value_segment`.